Build failed with openssl 1.1.1g

Hi, I've been trying to build nginx with this jdomain module but have been encountering the following errors.
Believe that it's caused by incompatibility with openssl 1.1.1.

-o nginx-1.18.0/addon/src/ngx_http_upstream_jdomain.o \
        /root/ngx_upstream_jdomain/src/ngx_http_upstream_jdomain.c
In file included from src/core/ngx_core.h:60,
                 from /root/ngx_upstream_jdomain/src/ngx_http_upstream_jdomain.c:8:
/root/ngx_upstream_jdomain/src/ngx_http_upstream_jdomain.c: In function ‘ngx_http_upstream_set_jdomain_peer_session’:
/root/ngx_upstream_jdomain/src/ngx_http_upstream_jdomain.c:605:42: error: dereferencing pointer to incomplete type ‘SSL_SESSION’ {aka ‘struct ssl_session_st’}
  605 |                 ssl_session ? ssl_session->references : 0);
      |                                          ^~
src/core/ngx_log.h:93:48: note: in definition of macro ‘ngx_log_debug’
   93 |         ngx_log_error_core(NGX_LOG_DEBUG, log, __VA_ARGS__)
      |                                                ^~~~~~~~~~~
/root/ngx_upstream_jdomain/src/ngx_http_upstream_jdomain.c:600:2: note: in expansion of macro ‘ngx_log_debug2’
  600 |  ngx_log_debug2(NGX_LOG_DEBUG_HTTP,
      |  ^~~~~~~~~~~~~~
make[1]: *** [nginx-1.18.0/Makefile:1578: nginx-1.18.0/addon/src/ngx_http_upstream_jdomain.o] Error 1
make[1]: Leaving directory '/root/nginx-1.18.0'
make: *** [Makefile:8: build] Error 2

[
![image](https://user-images.githubusercontent.com/70423998/91629121-ae151580-e9f8-11ea-9254-75ca0304d9df.png)
](url)

--- a/.github/actions/nginx-module-toolbox/Dockerfile
+++ b/.github/actions/nginx-module-toolbox/Dockerfile
@@ -1,3 +1,7 @@
 FROM nicholaschiasson/nginx-module-toolbox:amazonlinux-2-1.18.0-1
 
+WORKDIR /src/openssl
+RUN curl -sL https://www.openssl.org/source/openssl-1.1.1g.tar.gz -o openssl.tar.gz
+RUN tar --strip-components=1 -xzf openssl.tar.gz
+
 WORKDIR /github/workspace

--- a/scripts/build.sh
+++ b/scripts/build.sh
@@ -18,8 +18,6 @@ do
 	./configure \
 		--prefix=${BIN_DIR} \
 		--with-debug \
-		--with-http_ssl_module \
-		--with-openssl=/src/openssl \
 		--without-http_charset_module \
 		--without-http_userid_module \
 		--without-http_auth_basic_module \
@@ -36,7 +34,9 @@ do
 		--without-http_empty_gif_module \
 		--without-http_browser_module \
 		--without-http_upstream_ip_hash_module \
-		--add${TYPE}module=${GITHUB_WORKSPACE}
+		--add${TYPE}module=${GITHUB_WORKSPACE} \
+		# --with-http_ssl_module \
+		# --with-openssl=/src/openssl
 
 	make
 

--- a/scripts/build.sh
+++ b/scripts/build.sh
@@ -18,6 +18,8 @@ do
 	./configure \
 		--prefix=${BIN_DIR} \
 		--with-debug \
+		--with-http_ssl_module \
+		--with-openssl=/src/openssl \
 		--without-http_charset_module \
 		--without-http_userid_module \
 		--without-http_auth_basic_module \

--- a/src/ngx_http_upstream_jdomain.c
+++ b/src/ngx_http_upstream_jdomain.c
@@ -597,12 +597,8 @@ ngx_http_upstream_set_jdomain_peer_session(ngx_peer_connection_t* pc,
 
 	rc = ngx_ssl_set_session(pc->connection, ssl_session);
 
-	ngx_log_debug2(NGX_LOG_DEBUG_HTTP,
-	               pc->log,
-	               0,
-	               "set session: %p:%d",
-	               ssl_session,
-	               ssl_session ? ssl_session->references : 0);
+	ngx_log_debug1(
+	  NGX_LOG_DEBUG_HTTP, pc->log, 0, "set session: %p", ssl_session);
 
 	return rc;
 }
@@ -622,12 +618,8 @@ ngx_http_upstream_save_jdomain_peer_session(ngx_peer_connection_t* pc,
 		return;
 	}
 
-	ngx_log_debug2(NGX_LOG_DEBUG_HTTP,
-	               pc->log,
-	               0,
-	               "save session: %p:%d",
-	               ssl_session,
-	               ssl_session->references);
+	ngx_log_debug1(
+	  NGX_LOG_DEBUG_HTTP, pc->log, 0, "save session: %p", ssl_session);
 
 	peer = &urpd->conf->peers[urpd->current];
 
@@ -635,12 +627,8 @@ ngx_http_upstream_save_jdomain_peer_session(ngx_peer_connection_t* pc,
 	peer->ssl_session = ssl_session;
 
 	if (old_ssl_session) {
-		ngx_log_debug2(NGX_LOG_DEBUG_HTTP,
-		               pc->log,
-		               0,
-		               "old session: %p:%d",
-		               old_ssl_session,
-		               old_ssl_session->references);
+		ngx_log_debug1(
+		  NGX_LOG_DEBUG_HTTP, pc->log, 0, "old session: %p", old_ssl_session);
 
 		ngx_ssl_free_session(old_ssl_session);
 	}

